app/main.py

```diff
@@ -0,0 +1,178 @@
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+    # Include columns
+    include_columns = ["company_name", "scope_category", "time_frame", "temperature_score"] + \
+                      [column for column in include_columns if column in scores.columns]
+
+    return ResponseTemperatureScore(
+        aggregated_scores=aggregations,
+        scores=scores.where(pd.notnull(scores), None).to_dict(orient="records"),
+        coverage=coverage,
+        companies=scores[include_columns].replace({np.nan: None}).to_dict(orient="records"),
+        feature_distribution=column_distribution,
+    )
+
+
+class ResponseDataProvider(BaseModel):
+    name: str
+    type: str
+
+
+@app.get("/data_providers/", response_model=List[ResponseDataProvider])
+def get_data_providers() -> List[ResponseDataProvider]:
+    """
+    Get a list of the available data providers.
+    """
+    return [ResponseDataProvider(name=data_provider["name"], type=data_provider["type"])
+            for data_provider in config["data_providers"]]
+
+
+@app.post("/parse_portfolio/")
+def parse_portfolio(file: bytes = File(...), skiprows: int = Form(...)):
+    """
+    Parse a portfolio Excel file.
+
+    *Note: This endpoint is only for use in the frontend*
+    """
+    df = pd.read_excel(file, skiprows=int(skiprows))
+
+    return {'portfolio': df.replace(r'^\s*$', np.nan, regex=True).dropna(how='all').replace({np.nan: None}).to_dict(
+            orient="records")}
+
+
+@app.post("/import_data_provider/")
+def import_data_provider(file: UploadFile = File(...)):
+    """
+    Import a new Excel data provider file. This will overwrite the current "dummy" data provider input file.
+
+    *Note: This endpoint is only for use in the frontend during the beta testing phase.*
+    """
+    # TODO: Remove this endpoint after the beta testing phase
+    file_name = file.filename
+    file_type = file_name.split('.')[-1]
+    if file_type == 'xlsx':
+        x = 10000000 / 10000
+        xi = 0
+        with open(os.path.join(UPLOAD_FOLDER, 'InputFormat_tmp.xlsx'), 'ab') as f:
+            for chunk in iter(lambda: file.file.read(10000), b''):
+                f.write(chunk)
+                xi += 1
+                if xi > x:
+                    f.close()
+                    os.remove(os.path.join(UPLOAD_FOLDER, 'InputFormat_tmp.xlsx'))
+                    return {'POST Request': {'Response': {'Status Code': 400, 'Message': 'Error. File did not save.'}}}
+        os.rename(os.path.join(UPLOAD_FOLDER, 'InputFormat_tmp.xlsx'),
+                  os.path.join(UPLOAD_FOLDER, 'InputFormat.xlsx'))
+        return {'POST Request': {'Response': {'Status Code': 200, 'Message': 'Data Provider Imported'}}}
+    else:
+        return {'POST Request': {'Response': {'Status Code': 400, 'Message': 'Error. File did not save.'}}}
+
+
+if __name__ == "__main__":
+    uvicorn.run("main:app", host="127.0.0.1", port=5000, log_level="info", reload=True)
```

Comment:
port should probably be part of the config or env variable rather than hardcoded (same for log_level and reload)